src/main/java/software/amazon/awssdk/crt/auth/credentials/ProfileCredentialsProvider.java

```diff
@@ -0,0 +1,165 @@
+
+        long nativeHandle = profileCredentialsProviderNew(
+                this,
+                toNativeHandle(builder.clientBootstrap),
+                toNativeHandle(builder.tlsContext),
+                toByteArray(builder.profileName),
+                toByteArray(builder.configFileNameOverride),
+                toByteArray(builder.credentialsFileNameOverride)
+        );
+
+        acquireNativeHandle(nativeHandle);
+        if (builder.clientBootstrap != null) { addReferenceTo(builder.clientBootstrap); }
+        if (builder.tlsContext != null) { addReferenceTo(builder.tlsContext); }
+    }
+
+    /**
+     * A builder for creating a custom profile credentials provider.
+     */
+    public interface Builder {
+        /**
+         * Sets the client bootstrap (host resolver and event loop group) to use when making the connections
+         * required by this provider. The default is a bootstrap which uses the static default event loop group and host
+         * resolver.
+         * @param clientBootstrap client bootstrap to use
+         * @return The current builder
+         */
+        Builder withClientBootstrap(ClientBootstrap clientBootstrap);
+
+        /**
+         * Sets the tls context to use for any secure network connections made while sourcing credentials.
+         * @param tlsContext the tls context to use when establishing network connections
+         * @return The current builder
+         */
+        Builder withTlsContext(TlsContext tlsContext);
+
+        /**
+         * Sets the name of the profile to use. If none is specified, the profile named "default" is used.
+         * @param profileName the profile name to use
+         * @return The current builder
+         */
+        Builder withProfileName(String profileName);
+
+        /**
+         * Sets the name of the config file to use. If none is specified, a name of "~/.aws/config" (Linux and Mac) or
+         * "%USERPROFILE%\.aws\config" (Windows) is used.
+         * @param configFileNameOverride the config file name to use
+         * @return The current builder
+         */
+        Builder withConfigFileNameOverride(String configFileNameOverride);
+
+        /**
+         * Sets the name of the credentials file to use. If none is specified, a name of "~/.aws/credentials" (Linux and
+         * Mac) or "%USERPROFILE%\.aws\credentials" (Windows) is used.
+         * @param credentialsFileNameOverride the credentials file name to use
+         * @return The current builder
+         */
+        Builder withCredentialsFileNameOverride(String credentialsFileNameOverride);
+
+        /**
+         * Create a profile credentials provider using the configuration applied to this builder.
+         * @return A new profile credentials provider.
+         */
+        ProfileCredentialsProvider build();
+    }
+
+    static final class BuilderImpl implements Builder {
+        private ClientBootstrap clientBootstrap = new ClientBootstrap(null, null);
```

Comment:
Do this at provider construction time not as a default initializer.  As it stands, instantiating this class will always cause an event loop group and host resolver to be created lazily, which isn't desirable.

Reply:
Will do!